src/towncrier/check.py

```diff
@@ -52,27 +52,35 @@ def __main(comparewith, directory, config):
         click.echo(e.output)
         raise
 
-    if not files_changed:
+    # Convert changed files to a list.
+    files_changed = files_changed.split(os.linesep) if files_changed else []
+
+    # Create a set of changed files converted to absolute paths.
+    files = set(os.path.join(base_directory, f) for f in files_changed)
+
+    # Filter out ignored files.
+    ignore_files = [os.path.join(base_directory, f) for f in config["check_ignore_files"]]
```

Comment:
Wouldn't it be better to generate a set here?

Reply:
Agreed. Done.